src/utils/components/ResizableWrapper.tsx

```diff
@@ -41,17 +58,52 @@ export const ResizableWrapper = ({
     if (isResizing) {
       document.addEventListener('mousemove', onMouseMove);
       document.addEventListener('mouseup', onMouseUp);
+      scrollParent?.addEventListener('mouseenter', onScrollEnter);
+      scrollParent?.addEventListener('mouseleave', onScrollLeave);
     }
     return () => {
       document.removeEventListener('mousemove', onMouseMove);
       document.removeEventListener('mouseup', onMouseUp);
+      scrollParent?.removeEventListener('mouseenter', onScrollEnter);
+      scrollParent?.removeEventListener('mouseleave', onScrollLeave);
     };
-  }, [isResizing, currentWidth, clientX, deltaX, onWidthChange]);
+  }, [
+    isResizing,
+    currentWidth,
+    clientX,
+    deltaX,
+    onWidthChange,
+    scrollParent,
+    exitedScrollParent,
+  ]);
+
+  function getScrollParent(node: HTMLDivElement) {
+    const isElement = node instanceof HTMLDivElement;
+    const overflowX = isElement && window.getComputedStyle(node).overflowX;
+    const isScrollable = overflowX === 'scroll';
+
+    if (node.parentElement == null) {
+      return null;
+    }
+    if (isScrollable && node.scrollHeight > node.clientHeight) {
+      return node;
+    }
+    return getScrollParent(node.parentElement as HTMLDivElement);
+  }
```

Comment:
What is this recursive fn for? Why does scrollHeight matter? Don't we know the fix number of parents to traverse to get the `scrollParent`? To determine if the scroll is being activated or not, can't we just use the scrollWidth logic above?

Reply:
We do know where the scrollParent is, but there's a few wrappers spread between a few component files in between this ref and there, so I thought it might be a good idea to recursively grab it, in case anyone ends up adding another wrapper/component or reorganizing anything. Since the operation should only be happening on the first render, performance impact should be pretty minimal, though I can make it traverse a fixed number of parents upwards instead if you'd like.

To be honest the scrollHeight probably isn't necessary here, so I'll remove that one, ty for the catch.

I don't think we'd be able to use the scrollWidth logic from above for this, I'm pretty sure that width includes even the parts of the table that you can't see through the viewport of the scrollArea, though I wouldn't be surprised if there were other ways to determine whether the scroll was being activated or not, I only used the scrollParent for it since I had already gone through the effort of getting it and thought it might be the best way to do it haha.

Reply:
By using `scrollWidth` i mean `const isScrollable = scrollWidth === clientWidth`. Basically if the div is overflowing, the scrollwidth is bigger than the clientWidth.

Reply:
It may be the case that no scroll area is added, since we a dev can do `scrollable={false}`; so this may become an issue.

Reply:
The scrollAreaRef is now stored in the modelTableStore to handle this, additionally have set the resize to always cause scrolling (no longer dependent on leaving the scroll area).